code/game/objects/structures/railings.dm

```diff
@@ -183,27 +183,37 @@
 		if(4, 8)
 			layer = RAILING_MID_LAYER
 
-/obj/structure/railing/verb/revrotate()
+/obj/structure/railing/AltClick(mob/user)
+	if(user.incapacitated() || !Adjacent(user))
+		return
+	ccwrotate()
+
+/obj/structure/railing/verb/cwrotate()
 	set name = "Rotate Railing Clockwise"
 	set category = "Object"
 	set src in oview(1)
 
-	if(anchored)
-		to_chat(usr, "<span class='warning'>\The [src] is fastened to the floor, therefore you can't rotate it!</span>")
-		return
-
-	change_dir(turn(dir, 270))
+	rotate(270)
 
-/obj/structure/railing/verb/rotate()
+/obj/structure/railing/verb/ccwrotate()
 	set name = "Rotate Railing Counter-Clockwise"
 	set category = "Object"
 	set src in oview(1)
 
+	rotate(90)
+
+/obj/structure/railing/proc/rotate(var/angle)
```

Comment:
While it isn't really necessary, I'd recommend setting angle to a default value such as 90 in case the proc is directly called by mistake or by min.

Reply:
done